src/utils/slpReader.js

```diff
@@ -459,25 +461,39 @@ function readBool(view: DataView, offset: number): boolean | null {
   return !!view.getUint8(offset);
 }
 
-export function getMetadata(slpFile: SlpFileType): MetadataType {
+export function getMetadata(slpFile: SlpFileType): MetadataType | null {
   if (slpFile.metadataLength <= 0) {
     // This will happen on a severed incomplete file
     // $FlowFixMe
-    return {};
+    return null;
   }
 
   const buffer = new Uint8Array(slpFile.metadataLength);
 
   readRef(slpFile.ref, buffer, 0, buffer.length, slpFile.metadataPosition);
 
-  let metadata = {};
+  let metadata = null;
   try {
     metadata = decode(buffer);
   } catch (ex) {
     // Do nothing
-    console.log(ex);
+    // console.log(ex);
   }
 
   // $FlowFixMe
   return metadata;
 }
+
+export function getGameEnd(slpFile: SlpFileType): GameEndType | null {
+  if (slpFile.metadataLength <= 0) {
+    // This will happen on a severed incomplete file
+    // $FlowFixMe
+    return null;
+  }
+
+  const gameEndLength = slpFile.messageSizes[0x39] + 1;
+  const buffer = new Uint8Array(gameEndLength);
+  const start = slpFile.metadataPosition - METADATA_OFFSET - gameEndLength;
+  readRef(slpFile.ref, buffer, 0, gameEndLength, start);
+  return parseMessage(buffer[0], buffer);
```

Comment:
If buffer[0] is not 0x39 should we even call this function? Maybe just return null in that case? I'm slightly worried about garbage data having a command that exists in the first slot and the thing being processed weird.